admin-base/ui.apps/src/main/js/apiImpl.js

```diff
@@ -631,54 +631,75 @@ class PerAdminImpl {
-                    var extensionPart  = ""
-                    var indexOfLasDot = file.name.lastIndexOf('.');
-                    var newFileName
-                    if (indexOfLasDot > 0) {
-                        // filename has a dot
-                        localNamePart = file.name.substring(0, indexOfLasDot);
-                        extensionPart = file.name.substring(indexOfLasDot, file.name.length);
+            if (files.length == 1) {
+                $perAdminApp.askUser("File exists", "Select to replace the existing one, or keep both", {
+                    yesText:"Replace",
+                    noText:"Keep both",
+                    yes() {
+                        var $api = $perAdminApp.getApi()
+                        logger.info("user selected 'replace' upload file "+ file.name)
+                        var replaceData = new FormData()
+                        replaceData.append(file.name, file, file.name)
+                        return updateWithFormAndConfig('/admin/uploadFiles.json' + path, replaceData, config)
+                            .then(() => $api.populateNodesForBrowser(path))
+                            .catch( error => {
+                                log.error('Failed to upload: ' + error)
+                                reject('Unable to upload due to an error. '+ error)
+                            })
+                    },
+                    no() {
+                        logger.info("user selected 'keep both' make the uploaded file name unique and upload")
+                        var $api = $perAdminApp.getApi()
+                        var localNamePart = file.name
+                        var extensionPart  = ""
+                        var indexOfLasDot = file.name.lastIndexOf('.');
+                        var newFileName
+                        if (indexOfLasDot > 0) {
+                            // filename has a dot
+                            localNamePart = file.name.substring(0, indexOfLasDot);
+                            extensionPart = file.name.substring(indexOfLasDot, file.name.length);
+                        }
+                        let i = 1
+                        do {
+                          newFileName = localNamePart + i++ + extensionPart
+                        } while (!$api.nameAvailable(newFileName, path));
+                        var keepbothData = new FormData()
+                        keepbothData.append(newFileName, file, newFileName)
+                        return updateWithFormAndConfig('/admin/uploadFiles.json' + path, keepbothData, config)
+                            .then(() => $api.populateNodesForBrowser(path))
+                            .catch( error => {
+                                log.error('Failed to upload: ' + error)
+                                reject('Unable to upload due to an error. '+ error)
+                            })
                     }
-                    let i = 1
```

Comment:
Is using `let` and `const fine`? Or do you want everyone to only use `var` ?

Reply:
`var` is outdated and should not be used anymore since its not scope-save

use `let` for variables that are to be changed
use `const` for variables that are constant and never to be changed after defining them

Reply:
Ok. I think properties of constant objects can be changed, but primitive types cannot be changed if const.

Reply:
Exactly. Same for array.